cypress/e2e/group_test.spec.ts

```diff
@@ -1,219 +1,459 @@
+      cy.reload();
+      groupPage
+        .selectGroupItemCheckboxAllRows()
+        .deleteSelectedGroups()
+        .assertNotificationGroupsDeleted()
+        .assertNoGroupsInThisRealmEmptyStateMessageExist(true);
+    });
+  });
 
-      sidebarPage.waitForPageLoad();
-      groupModal
-        .open("openCreateGroupModal")
-        .fillGroupForm(targetGroupName)
-        .clickCreate();
+  describe("Search group under current group", () => {
+    before(async () => {
+      const createdGroups = await adminClient.createSubGroups(predefinedGroups);
+      for (let i = 0; i < 5; i++) {
+        adminClient.addUserToGroup(users[i].id!, createdGroups[i % 3].id);
+      }
+      adminClient.createUser({ username: "new", enabled: true });
+    });
 
-      // For some reason, this fixes clickDetailMenu
-      sidebarPage.goToEvents();
-      sidebarPage.goToGroups();
+    it("Search child group in group", () => {
+      groupPage
+        .goToGroupChildGroupsTab(predefinedGroups[0])
+        .searchGroup(predefinedGroups[1])
+        .goToGroupChildGroupsTab(predefinedGroups[1])
+        .searchGroup(predefinedGroups[2])
+        .assertGroupItemExist(predefinedGroups[2], true);
+    });
 
-      listingPage.clickRowDetails(groupName).clickDetailMenu("Move to");
-      moveGroupModal
-        .clickRow(targetGroupName)
-        .checkTitle(`Move ${groupName} to ${targetGroupName}`);
-      moveGroupModal.clickMove();
-
-      masthead.checkNotificationMessage("Group moved");
-      sidebarPage.waitForPageLoad();
-      listingPage.itemExist(groupName, false);
-      clickGroup(targetGroupName);
-      sidebarPage.waitForPageLoad();
-      listingPage.itemExist(groupName);
-      sidebarPage.goToGroups();
-      listingPage.deleteItem(targetGroupName);
-      modalUtils.checkModalTitle("Delete group?").confirmModal();
-      masthead.checkNotificationMessage("Group deleted");
+    it("Search non existing child group in group", () => {
+      groupPage
+        .goToGroupChildGroupsTab(predefinedGroups[0])
+        .searchGroup("non-existent-sub-group")
+        .assertNoSearchResultsMessageExist(true);
     });
 
-    describe("Move groups", () => {
-      const groups = ["group", "group1", "group2"];
-      before(() => adminClient.createSubGroups(groups));
+    it("Empty search in group", () => {
+      groupPage
+        .goToGroupChildGroupsTab(predefinedGroups[0])
+        .searchGroup("   ")
+        .assertNoSearchResultsMessageExist(true);
+    });
+  });
 
-      after(() => adminClient.deleteGroups());
+  describe("Group Actions", () => {
+    const groupNameDeleteHeaderAction = "group_test_delete_header_action";
 
-      it("Should move group to root", () => {
-        listingPage.goToItemDetails(groups[0]);
-        sidebarPage.waitForPageLoad();
-        listingPage.clickRowDetails(groups[1]).clickDetailMenu("Move to");
+    before(async () => {
+      await adminClient.createGroup(groupNameDeleteHeaderAction);
+    });
```

Comment:
You can return the Promise here instead of awaiting it.
```suggestion
    before(() => adminClient.createGroup(groupNameDeleteHeaderAction));
```